### PR TITLE
Add `RelativeSquaredError`

### LINKED ISSUE #1764: Add `torchmetrics.regression.RelativeSquaredError`
## 🚀 Feature

Add a regression metric for the (mean) relative squared error, defined as the sum of the squared error, normalized by the squared deviation of the targets from the mean.

### Motivation

The relative squared error is a useful metric for comparing regressions across time series and problems, as it is scale-invariant. This makes it a superior metric to the mean squared error in a number of cases, such as when a consistent metric is required for defining the performance of a particular model type across different datasets.

### Pitch

We desire a functional and Module-like implementation of the relative squared error. The implementation is rather straightforward; in fact, the `torchmetrics`-style metric update function is identical to `torchmetrics.functional.regression.r2._r2_score_update`. I will provide a PR shortly.

### PR DESCRIPTION
**Pull request by wbeardall**
## What does this PR do?

Fixes #1764
This PR provides functional and Module-style implementations for the scale-invariant `RelativeSquaredError` metric.

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/torchmetrics/functional/regression/rse.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    rse = sum_squared_error / torch.clamp((sum_squared_obs - sum_obs * sum_obs / n_obs), min=epsilon)
+    if not squared:
+        rse = torch.sqrt(rse)
+    return torch.mean(rse)
+
+
+def relative_squared_error(preds: Tensor, target: Tensor, squared: bool = True) -> Tensor:
+    r"""Computes the relative squared error (RSE).
+
+    .. math:: \text{RSE} = \frac{\\sum_i^N(y_i - \\hat{y_i})^2}{\\sum_i^N(y_i - \\overline{y})^2}
```
Comment (Borda) [inline src/torchmetrics/functional/regression/rse.py:45]:
```suggestion
    .. math:: \text{RSE} = \frac{\\sum_i^N(y_i - \\hat{y_i})^2}{\\sum_i^N(y_i - \\overline{y})^2}
    
```

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/functional/regression/rse.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+from torchmetrics.functional.regression.r2 import _r2_score_update
+
+
+def _relative_squared_error_compute(
+    sum_squared_obs: Tensor,
+    sum_obs: Tensor,
+    sum_squared_error: Tensor,
+    n_obs: Union[int, Tensor],
+    squared: bool = True,
+    epsilon: float = 1.17e-06,
```
Comment (justusschock) [inline src/torchmetrics/functional/regression/rse.py:15]:
```suggestion
```

File: src/torchmetrics/functional/regression/rse.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    Example:
+        >>> target = torch.tensor([[0.5, 1], [-1, 1], [7, -6]])
+        >>> preds = torch.tensor([[0, 2], [-1, 2], [8, -5]])
+        >>> # RSE uses the same update function as R2 score.
+        >>> sum_squared_obs, sum_obs, rss, n_obs = _r2_score_update(preds, target)
+        >>> _relative_squared_error_compute(sum_squared_obs, sum_obs, rss, n_obs, squared=True)
+        tensor(0.0632)
+    """
+    rse = sum_squared_error / torch.clamp(sum_squared_obs - sum_obs * sum_obs / n_obs, min=epsilon)
```
Comment (justusschock) [inline src/torchmetrics/functional/regression/rse.py:36]:
```suggestion
    epsilon = torch.finfo(sum_squared_error).eps
    rse = sum_squared_error / torch.clamp(sum_squared_obs - sum_obs * sum_obs / n_obs, min=epsilon)
```

**Review by justusschock — COMMENTED** [review]
LGTM in general, two minor suggestions :)

**Review by justusschock — COMMENTED** [review]
LGTM in general, two minor suggestions :)

**Review by justusschock — APPROVED** [review]

File: src/torchmetrics/functional/regression/rse.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    """Computes Relative Squared Error.
+
+    Args:
+        sum_squared_obs: Sum of square of all observations
+        sum_obs: Sum of all observations
+        sum_squared_error: Residual sum of squares
+        n_obs: Number of predictions or observations
+        squared: Returns RRSE value if set to False.
+        epsilon: Specifies the lower bound for target values. Any target value below epsilon
+            is set to epsilon (avoids ``ZeroDivisionError``).
```
Comment (SkafteNicki) [inline src/torchmetrics/functional/regression/rse.py:38]:
```suggestion
```

**Review by SkafteNicki — APPROVED** [review]
